cli/flags.go

```diff
@@ -433,6 +481,14 @@ func init() {
 			visitFlags(cmd, base.AddToFlagMap)
 		})
 
+		// If any of the security flags have been set, clear the insecure
+		// setting. Note that we do the inverse when the --insecure flag is
+		// set. See insecureValue.Set().
+		if cliContext.SSLCA != "" || cliContext.SSLCAKey != "" ||
+			cliContext.SSLCert != "" || cliContext.SSLCertKey != "" {
+			cliContext.Insecure = false
+		}
+
 		cliContext.Addr = net.JoinHostPort(connHost, connPort)
```

Comment:
Why is the part in `start.go` not necessary? If `--insecure` isn't specified, we want to only listen on localhost, not the empty host.